### PR TITLE
[WIP] Enabling --request-header-ca to be passed to the api server

### LINKED ISSUE #1172: API server crashes when installed on Minikube
After installing service-catalog on Minikube running Kubernetes v1.7.0 (see [install doc](https://github.com/kubernetes-incubator/service-catalog/blob/master/docs/install-1.7.md)), the API server crashes with the following log message:

```console
F0829 23:19:24.357279       1 server.go:128] error running server (cluster doesn't provide requestheader-client-ca-file)
```

### PR DESCRIPTION
**Pull request by arschles**
Fixes #1172

According to [docs on certs & auth](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/concepts/auth.md#requestheader-authentication), if the aggregator (i.e. the core Kube API server) is not configured with `--requestheader-client-ca-file`, we have to pass that flag to the "addon API server", which is our API server.

Minikube by default does not start the core API server with that flag, so this PR enables passing that flag to the service-catalog API server.

Now that I'm able to get an API server running on Minikube, I'm running into the below issue:

```console
$ k get broker
Error from server (Forbidden): User "system:anonymous" cannot list brokers.servicecatalog.k8s.io at the cluster scope. (get brokers.servicecatalog.k8s.io)
```

(note that `servicebroker` won't work because the chart doesn't yet use an image that includes the resource name changes)

So, I've marked this PR as a work in progress while I investigate.

cc/ @DirectXMan12

### CODE REVIEW

File: docs/install-1.7.md [hunk 1]
```diff
@@ -156,4 +157,5 @@ helm install ../../charts/catalog \
         --set apiserver.tls.ca=$(base64 ${SC_SERVING_CA}) \
         --set apiserver.tls.cert=$(base64 ${SC_SERVING_CERT}) \
         --set apiserver.tls.key=$(base64 ${SC_SERVING_KEY})
```
Comment (krancour) [inline docs/install-1.7.md]: Need to add a trailing `\` on this line.

File: docs/install-1.7.md [hunk 2]
```diff
@@ -144,6 +144,7 @@ helm install ../../charts/catalog \
         --set apiserver.tls.ca=$(base64 --wrap 0 ${SC_SERVING_CA}) \
         --set apiserver.tls.cert=$(base64 --wrap 0 ${SC_SERVING_CERT}) \
         --set apiserver.tls.key=$(base64 --wrap 0 ${SC_SERVING_KEY})
```
Comment (krancour) [inline docs/install-1.7.md]: Trailing `\` needed.

Comment (arschles) [reply docs/install-1.7.md]: done

File: docs/install-1.7.md (hunk 1 above)
Comment (arschles) [reply docs/install-1.7.md]: done

File: docs/install-1.7.md [hunk 3]
```diff
@@ -143,7 +143,8 @@ helm install ../../charts/catalog \
         --set useAggregator=true \
         --set apiserver.tls.ca=$(base64 --wrap 0 ${SC_SERVING_CA}) \
         --set apiserver.tls.cert=$(base64 --wrap 0 ${SC_SERVING_CERT}) \
-        --set apiserver.tls.key=$(base64 --wrap 0 ${SC_SERVING_KEY})
+        --set apiserver.tls.key=$(base64 --wrap 0 ${SC_SERVING_KEY}) \
+        --set apiserver.tls.requestHeaderCA=$(base64 ${SC_SERVING_CA})
```
Comment (krancour) [inline docs/install-1.7.md]: Should this last command use `--wrap` like the others?

Comment (arschles) [reply docs/install-1.7.md]: yes, good catch

**Comment by nilebox** [comment]
> Now that I'm able to get an API server running on Minikube, I'm running into the below issue:

```
$ k get broker
Error from server (Forbidden): User "system:anonymous" cannot list brokers.servicecatalog.k8s.io at the cluster scope. (get brokers.servicecatalog.k8s.io)
```
@arschles try following the steps provided by @kibbles-n-bytes: https://github.com/kubernetes-incubator/service-catalog/issues/1069#issuecomment-319444019 (make sure that you have a correct `--server` value, mine are different from what Michael have provided.

**Comment by arschles** [comment]
There is a fix for minikube in progress [here](https://github.com/kubernetes/minikube/issues/1404) (the PR isn't up yet afaik) that will make this unnecessary for Minikube installations. We are also working on a similar fix for [acs-engine](https://github.com/azure/acs-engine)

**Comment by pmorie** [comment]
Looks like jenkins build flaked; I kicked it.

**Comment by pmorie** [comment]
@arschles is this ready to go in?

**Comment by pmorie** [comment]
Might have to be in 0.1.1 - @arschles is this still an issue/needed?

**Comment by arschles** [comment]
We don't need this PR if kubernetes installs have the right CA data in the `kube-system/extension-apiserver-authentication` config map (that configmap needs at least the `requestheader-client-ca-file` key). We've been focusing on making common kubernetes installs have that key instead, so I am closing this PR.

**Pull request closed without merging**